core/src/test/java/org/elasticsearch/index/query/AbstractQueryTestCase.java

```diff
@@ -390,6 +392,14 @@ public void testFromXContent() throws IOException {
         }
     }
 
+    /**
+     * subclasses should override this method in case some fields in xContent should be protected from random
+     * shuffling in the {@link #testFromXContent()} test case
+     */
+    protected Set<String> provideShuffleproofFields() {
```

Comment:
I also thought so, but at least for queries - with the exception of PercolatorQueryBuilder - random ordering of fields worked well. I ran the test suite several times, see what else CI brings up. But this way will detect those things.